### PR TITLE
DOCS: autoapi adoption for create API docs.

### PR DESCRIPTION
**Pull request by tkknight**
### Description

Adoption of [autoapi](https://github.com/readthedocs/sphinx-autoapi).

Notes:

* ``napolean`` is needed so the api docs render correctly
* Excluded the ``geovista/examples``
* ``show_toc_level`` set to `3`, so the righ hand "on page" bookmarks are shown.,  Useful for a large module which many functions, classes and methods.
* Couple of minor typos fixed in docstrings.

### TODO

- [x]  ``SPHIXOPTS`` comment to be reviewed.  If we can resolve the build warnings this can be enabled.
- [x] ``autoapi_options`` to be finalized

---

### CODE REVIEW

**Comment by tkknight** [comment]
pre-commit.ci autofix

File: docs/src/conf.py [hunk 1]
```diff
@@ -82,6 +106,35 @@
 # src base directory
 base_dir = Path(__file__).absolute().parent
 
+# -- autoapi extension --------------------------------------------------------
+# See https://sphinx-autoapi.readthedocs.io/en/latest/reference/config.html
+#     https://github.com/readthedocs/sphinx-autoapi
+#
+source_code_root = (Path(__file__).parents[2]).absolute()
+module_dir = source_code_root / "src"
```
Comment (bjlittle) [inline docs/src/conf.py:114]:
```suggestion
module_dir = root_dir / "src"
```

Comment (bjlittle) [inline docs/src/conf.py:114]: @tkknight Could you move the `root_dir` and `module_dir` to live with the `base_dir`, thanks 👍

File: docs/src/conf.py [hunk 2]
```diff
@@ -82,6 +106,35 @@
...
+    "show-inheritance",
+    "show-module-summary",
+    #'special-members',
+    "imported-members",
+]
+
+autoapi_python_class_content = "both"
+autoapi_keep_files = True
+
+autolog(f"[autoapi] {source_code_root = }")
```
Comment (bjlittle) [inline docs/src/conf.py:134]:
```suggestion
autolog(f"[autoapi] {root_dir = }")
```

**Comment by tkknight** [comment]
pre-commit.ci autofix

**Review by bjlittle — APPROVED** [review]
@tkknight This is great! Thanks so much for pushing this forwards 🚀🥳

**Comment by bjlittle** [comment]
@all-contributors please add @tkknight for doc and infra

**Pull request merged**